### DISCUSSION TITLE
Beginner here!

### DISCUSSION THREAD

**Discussion by VukMar** [codewars/codewars.com#2549] [Q&A] [answered]
I have a stupid question, but I started learning code on my own and I can't seam to understand this code

#include <string>
#include <sstream>

int find_short(std::string str)
{
    std::istringstream inp(str);
    std::string s;
    int len = -1;         //This line says the int is -1, right?
    while (std::getline(inp, s, ' ')) 
    if (s.length() < len) //And here we compare "-1" to the length of a string
    len = s.length();
    return len;
}
so how can if statement be true when length of the word can never be less then "-1"?
What am I reading wrong?

**Comment by amezents** [comment, marked as answer]
cppreference states: [std::string.size()](https://en.cppreference.com/w/cpp/string/basic_string/size)(and std::string.length()) is defined as std::distance(begin(), end()),  where  begin() and end() are first and last input  iterators.
[std::distance](https://en.cppreference.com/w/cpp/iterator/distance) is number of hops from first to last - positive number (positive direction is movement to right from lesser addresses to greater) . The value may be negative if random-access iterators are used and first is reachable from last. This means that me moves to the left (negative direction) from greater addresses to lesser  So s.length() can be less then -1.
